src/bindings/python/src/pyopenvino/core/async_infer_queue.cpp

```diff
@@ -21,71 +21,85 @@ namespace py = pybind11;
 
 class AsyncInferQueue {
 public:
-    AsyncInferQueue(std::vector<InferRequestWrapper> requests,
-                    std::queue<size_t> idle_handles,
-                    std::vector<py::object> user_ids)
-        : _requests(requests),
-          _idle_handles(idle_handles),
-          _user_ids(user_ids) {
+    AsyncInferQueue(ov::CompiledModel& model, size_t jobs) {
+        if (jobs == 0) {
+            jobs = (size_t)Common::get_optimal_number_of_requests(model);
+        }
+
+        for (size_t handle = 0; handle < jobs; handle++) {
+            // Create new "empty" InferRequestWrapper without pre-defined callback,
+            // inputs and outputs.
+            auto request = InferRequestWrapper(model.create_infer_request());
+            // Copy Inputs and Outputs from ov::CompiledModel
+            request.m_inputs = model.inputs();
+            request.m_outputs = model.outputs();
+
+            m_requests.push_back(request);
```

Comment:
Optionally, to be tested. Only possible if the InferRequestWrapper supports move semantics.
```suggestion
            m_requests.push_back(std::move(request));
```

Reply:
Finally replaced with `emplace_back` method, I will fallback to this in case something is wrong.